app/src/main/java/com/telnyx/webrtc/sdk/ui/wsmessages/WsMessageFragment.kt

```diff
@@ -0,0 +1,149 @@
+package com.telnyx.webrtc.sdk.ui.wsmessages
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.content.FileProvider
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.telnyx.webrtc.sdk.BuildConfig
+import com.telnyx.webrtc.sdk.R
+import com.telnyx.webrtc.sdk.databinding.FragmentWsmessageBinding
+import com.telnyx.webrtc.sdk.model.WsMessageData
+import com.telnyx.webrtc.sdk.ui.MainViewModel
+import com.telnyx.webrtc.sdk.verto.receive.SocketObserver
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileOutputStream
+
+
+class WsMessageFragment : Fragment() {
+    private lateinit var binding: FragmentWsmessageBinding
+    private val mainViewModel: MainViewModel by activityViewModels()
+    private var receivedWsMessageList = emptyList<String>()
+
+    private var wsMessageAdapter: WsMessageAdapter? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentWsmessageBinding.inflate(inflater, container, false)
+        receivedWsMessageList = arguments?.getStringArrayList(WSMESSAGES_LIST)!!
+
+        return binding.root
+    }
+
+    override fun onResume() {
+        super.onResume()
+        setAdapter()
+        observeWsMessages()
+        setupViews()
+    }
+
+    private fun setAdapter() {
+        wsMessageAdapter =
+            WsMessageAdapter(
+                receivedWsMessageList.toMutableList()
+            )
+
+        binding.recyclerViewWsMessages.adapter = wsMessageAdapter
+        wsMessageAdapter?.let {
+            val lastPosition = it.itemCount - 1
+            if (lastPosition >= 0) {
+                binding.recyclerViewWsMessages.layoutManager =
+                    LinearLayoutManager(context).apply {
+                        scrollToPositionWithOffset(lastPosition, 0)
+                    }
+            }
+        }
+    }
+
+    private fun observeWsMessages() {
+        mainViewModel.getWsMessageResponse()?.observe(
+            this.viewLifecycleOwner, object : SocketObserver<WsMessageData>() {
+                override fun onConnectionEstablished() {}
+
+                override fun onMessageReceived(data: WsMessageData?) {
+                    data?.wsMessageJsonObject?.let {
+                        wsMessageAdapter?.addWsMessages(it.toString())
+                    }
+                }
+
+                override fun onLoading() {}
+
+                override fun onError(message: String?) {}
```

Comment:
See comment above, if we switch to the raw JsonObject we wouldn't have to override 3 methods we don't use here. 

Although if we keep this method then we should at least print a message of some kind when the other messages are received